### ISSUE TITLE
Using RFC2136 dns challenge does not allow to create multi domain certs.

### ISSUE THREAD

**Issue by middelink** [xenolf/lego#140]
Since the rfc2136 provider requires me to set the zone upfront, i cannot create a cert covering more than one zone. e.g. RFC2136_ZONE=a.nl, ./lego --dns=rfc2136 -d www.a.nl -d www.b.nl run

Since I realize this is rather dns-setup specific, could we perhaps create a zone with a special meaning? like "auto", which will derive the zone from the last 2 parts of the fqdn? aka www.a.nl > a.nl ?  Perhaps even parametrize it? auto:3 for say a.co.uk zones.

**Comment by janeczku** [comment]
I guess we could lookup the zone names for the domains like the other DNS providers do and get rid of the `RFC2136_ZONE` parameter all together.

**Comment by middelink** [comment]
That is ... awkward. I guess one could query for the SOA on each level till one finds a SOA.
bigbox.ch.polyware.nl > nothing
ch.polyware.nl > nothing
polyware.nl > SOA > gotcha

But yeah, for lack of a better way, that might be the best we can do.

**Comment by janeczku** [comment]
No need to traverse - the `authority section` contains the SOA of the zone:

> dig SOA bigbox.ch.polyware.nl @8.8.8.8

```SHELL
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60090
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;bigbox.ch.polyware.nl.		IN	SOA

;; AUTHORITY SECTION:
polyware.nl.		1169	IN	SOA	ns1.iaf.nl. netwerk.iaf.nl. 2016022606 28800 7200 2419200 1200
```

**Comment by middelink** [comment]
Are you sure all nameservers will report the authority section? Looking at various internet posting on the topic made me doubt that assumption. (Although I agree that if you are incapable of setting RC2136_NAMESERVER to *your own* authoritative nameserver, you probably should not be doing this at all.)

**Comment by janeczku** [comment]
Name servers authoritative for a zone MUST include the SOA record of
   the zone in the authority section of the response when reporting an
   NXDOMAIN or indicating that no data of the requested type exists

**Comment by middelink** [comment]
Yeah, that part was clear. Just that not any nameserver would do, so the query needs to be done against RFC2136_NAMESERVER.

I'll try to whip up a patch

**Comment by janeczku** [comment]
> so the query needs to be done against RFC2136_NAMESERVER

That's what i thought too. We can't expect a recursive nameserver to include the SOA in it's answer.

**Comment by middelink** [comment]
https://github.com/xenolf/lego/pull/143

**Comment by xenolf** [comment]
Closed per #143
